basic/http/collect-user.js

```diff
@@ -25,13 +29,35 @@ class User {
 };
 
 const user = new User();
+
 router.get( '/users', async ctx => {
-  ctx.body = {data:user.data.map((e)=>{
+  ctx.body = {
+    data:user.data.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  };
+});
+
+router.post('/users',async ctx => {
+  user.data.push(ctx.request.body);
+  
+  ctx.body = {
+    data:user.data.map((e)=>{
       return {id:e.id,name:e.name}; 
     })
   };
+});
 
+router.patch('/users',async ctx => {
```

Comment:
we need to get the id for `patch`. The routes should be like this:
```
router.patch('/users/:id' ...);
```

Please see the documentation of koa-router,  [docs](https://github.com/ZijianHe/koa-router)

Reply:
RESOLVED 💯